app/src/main/java/org/fossasia/openevent/app/core/settings/ScanSettings.java

```diff
@@ -0,0 +1,84 @@
+package org.fossasia.openevent.app.core.settings;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.preference.CheckBoxPreference;
+import android.support.v7.preference.PreferenceManager;
+import android.support.v7.preference.Preference.OnPreferenceChangeListener;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat;
+
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.common.Constants;
+import org.fossasia.openevent.app.ui.ViewUtils;
+
+public class ScanSettings extends PreferenceFragmentCompat {
+
+    public static ScanSettings newInstance() {
+        return new ScanSettings();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+        view.setBackgroundColor(getResources().getColor(R.color.color_top_surface));
+        return view;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis") // Inevitable DU Anomaly
+    public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
+        PreferenceManager manager = getPreferenceManager();
+        manager.setSharedPreferencesName(Constants.FOSS_PREFS);
+
+        setPreferencesFromResource(R.xml.scan_settings, rootKey);
+
+        CheckBoxPreference checkOut = (CheckBoxPreference) findPreference("check_out");
+        CheckBoxPreference validate = (CheckBoxPreference) findPreference("validate");
+        CheckBoxPreference checkIn = (CheckBoxPreference) findPreference("check_in");
+
+        OnPreferenceChangeListener listener = (preference, newValue) -> {
+            String key = preference.getKey();
+
+            switch (key) {
+                case "check_in":
+                    //Reset other items
+                    checkOut.setChecked(false);
+                    validate.setChecked(false);
+                    break;
+                case "check_out":
+                    //Reset other items
+                    checkIn.setChecked(false);
+                    validate.setChecked(false);
+                    break;
+                case "validate":
+                    //Reset other items
+                    checkOut.setChecked(false);
+                    checkIn.setChecked(false);
+                    break;
+                default:
+                    break;
+            }
+
+            //Force the current focused checkbox to always stay checked when pressed
+            //i.e confirms value when newValue is checked (true) and discards newValue
+            //when newValue is unchecked (false)
+            return (Boolean) newValue;
+        };
```

Comment:
Why not use radio buttons?

Reply:
Because probably radio buttons can not be used in preferences. We'd have to use ListPreferences, which won't give similar UI as this:
<img src="https://user-images.githubusercontent.com/35009811/41808127-b04f17e6-76f6-11e8-9a15-e1b453caec58.png" width="200"/>

Reply:
OK, what will happen on validate option?

Reply:
Validate would check if a ticket is valid. It won't check-in/out attendees.

Reply:
That's literal explanation. What would it do to validate is what I meant

Reply:
This preference setting will be used in `ScanQRPresenter.java`. It has a method called `processBarcode()`
```
private void processBarcode(String barcode) {
        getView().showBarcodeData(barcode);

        Observable.fromIterable(attendees)
            .compose(dispose(getDisposable()))
            .filter(attendee -> attendee.getOrder() != null)
            .filter(attendee -> (attendee.getOrder().getIdentifier() + "-" + attendee.getId()).equals(barcode))
            .compose(schedule())
            .subscribe(attendee -> getView().onScannedAttendee(attendee));
    }
```
Where we can appropriately react on what happens if a ticket is valid, like showing a snackbar. This change should probably be included after the scanning bug is fixed.